### PR TITLE
Upgrade plugins & dependencies November edition

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

**Comment by wendigo** [comment]
PTAL @electrum @martint

**Comment by wendigo** [comment]
Tested with airlift

**Comment by wendigo** [comment]
ping @martint @electrum

File: CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Airbase 149
 
 * Raise target JDK version to 17
+* Remove bval dependency
```
[the remark this answers is not in the document]
Comment (wendigo) [reply CHANGES.md:4]: It was replaced with hibernate validator when we moved from javax to jakarta dependencies. It's a cleanup

File: airbase/pom.xml [hunk 2]
```diff
@@ -162,33 +162,32 @@
...
         <dep.guice.version>7.0.0</dep.guice.version>
         <dep.jackson.version>2.15.3</dep.jackson.version>
         <dep.jmh.version>1.37</dep.jmh.version>
         <dep.jmxutils.version>1.24</dep.jmxutils.version>
         <dep.joda.version>2.12.5</dep.joda.version>
-        <dep.junit.version>5.10.0</dep.junit.version>
-        <dep.kotlin.version>1.9.0</dep.kotlin.version>
+        <dep.junit.version>5.10.1</dep.junit.version>
+        <dep.kotlin.version>1.9.21</dep.kotlin.version>
         <dep.logback.version>1.4.8</dep.logback.version>
```
Comment (nineinchnick) [inline airbase/pom.xml:181]: @wendigo can you also update this to 1.4.14? It fixes two CVEs: https://logback.qos.ch/news.html

Comment (wendigo) [reply airbase/pom.xml:181]: No. It's broken, hangs the execution in Trino.

Comment (wendigo) [reply airbase/pom.xml:181]: Already tested. There is a regressionm in 1.4.9 lurking somewhere

Comment (nineinchnick) [reply airbase/pom.xml:181]: Does it make sense to test 1.3.14?

Comment (wendigo) [reply airbase/pom.xml:181]: Already did that. Still failing. You can try it on your own

Comment (nineinchnick) [reply airbase/pom.xml:181]: It requires changes in Airlift: https://github.com/airlift/airlift/pull/1112

**Comment by wendigo** [comment]
@martint PTAL

**Review by nineinchnick — APPROVED** [review]

File: CHANGES.md [hunk 3]
```diff
@@ -1,7 +1,33 @@
...
+  - maven-clean-plugin 3.3.2 (from 3.3.1)
+  - maven-javadoc-plugin 3.6.2 (from 3.6.0) 
+  - maven-pmd-plugin 3.21.2 (from 3.21.0)
+  - maven-release-plugin 3.0.1 (from 2.5)
+  - maven-source-plugin to 3.3.0 (from 3.0.1)
+  - maven-surefire-plugin to 3.2.2 (from 3.2.1)
+  - build-helper-plugin to 3.5.0 (from 3.4.0)
+* Dependency updates:
+  - spotbugs-annotations 4.8.1 (from 4.8.0)
+  - errorprone-annotations 2.23.0 (from 2.22.0)
```
Comment (nineinchnick) [inline CHANGES.md:17]:
```suggestion
  - errorprone-annotations 2.24.1 (from 2.22.0)
```

Comment (nineinchnick) [reply CHANGES.md:17]: Oh there's another line for errorprone below

File: airbase/pom.xml [hunk 4]
```diff
@@ -51,7 +51,7 @@
     <properties>
         <!-- see http://docs.codehaus.org/display/MAVENUSER/POM+Element+for+Source+File+Encoding -->
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <project.build.targetJdk>8</project.build.targetJdk>
+        <project.build.targetJdk>17</project.build.targetJdk>
```
Comment (martint) [inline airbase/pom.xml:54]: Why 17?

Comment (wendigo) [reply airbase/pom.xml:54]: Preparation for maven 4.0.0

Comment (martint) [reply airbase/pom.xml]: I guess what I’m asking is why not 21?

Comment (wendigo) [reply airbase/pom.xml]: I've dropped this commit. We can revisit it later.

File: CHANGES.md [hunk 5]
```diff
@@ -13,7 +13,8 @@ Airbase 149
 * Dependency updates:
   - opentelemetry to 1.34.0 (from 1.31.0)
   - opentelemetry-instrumentation to 1.32.0 (from 1.31.0)
-  - kotlin 1.9.22 (from 1.9.0)
+  - kotlin 1.9.21 (from 1.9.0)
```
Comment (martint) [inline CHANGES.md:17]: Unintended change?

Comment (wendigo) [reply CHANGES.md]: done

**Pull request merged**